include/ruby/internal/intern/variable.h

```diff
@@ -39,6 +39,15 @@ RBIMPL_SYMBOL_EXPORT_BEGIN()
  */
 VALUE rb_mod_name(VALUE mod);
 
+/**
+ * Sets the anonymous name of a module.
+ * 
+ * @param[in]  mod        An instance of ::rb_cModule.
+ * @param[in]  name       An instance of ::rb_cString.
+ * @retval     mod
+ */
+VALUE rb_mod_name_set(VALUE, VALUE);
```

Comment:
Does this function need to be exposed?

Reply:
I just added it next to `rb_mod_name` for consistency, but we don't have to.

I don't have a strong opinion about it. I can imagine some use cases.

Reply:
I think better not to expose it, one can just use `rb_funcall` if they want to call it from C.
Also this function name does not include temporary or anything, so it seems better to stay an internal impl detail.

Reply:
It's quite messy to optionally detect something that must be done at runtime, vs `have_func("rb_mod_name_set")`. I don't have a particular preference about the name, but I think the proposed name is consistent with the other names. However, if you have a better suggestion, I'm happy to adopt it.

Reply:
> It's quite messy to optionally detect something that must be done at runtime using rb_funcall

It's not, there is `rb_respond_to()` and `rb_check_funcall()`.

I think it's obvious that rb_mod_name_set would be the (unlimited) setter and rb_mod_name the getter. But that's not case. There is no reason to expose this C API function and it has a cost.

Reply:
Okay, I'll rename it.

Reply:
If the function is exposed in Ruby, why are you so strongly against exposing it as part of the C API? Is it because it's difficult for compatibility with TruffleRuby's C interface layer?

I'm okay to hide it for now, but I may introduce it if I need it.

Reply:
> why are you so strongly against exposing it as part of the C API?

* I think the function name is bad/confusing, there is no need to discuss it if we don't add it
* There is no need to add it
* There is a cost to expose a C API function, it needs to be tested (e.g. C API tests) and reimplemented on other Ruby implementations. IMO we should not expose something which does exactly the same as `rb_funcall`, that's duplication (unless it makes a big difference in performance maybe, not relevant here).